src/main/steps/applicant/support-you-need-during-case/SupportYouNeedDuringCaseController.ts

```diff
@@ -17,60 +19,40 @@ export class SupportYouNeedDuringYourCaseController extends PostController<AnyOb
   }
   public async post(req: AppRequest, res: Response): Promise<void> {
     try {
-      const caseworkerUser = req.session.user;
-      const caseReference = req.session.userCase.id;
-
-      const client = new CosApiClient(caseworkerUser.accessToken, 'https://return-url');
-
-      const caseDataFromCos = await client.retrieveByCaseId(caseReference, caseworkerUser);
-      Object.assign(req.session.userCase, caseDataFromCos);
-
-      if (req.url.includes('respondent') || req.url.includes('tasklistresponse')) {
-        if ('C100' === req.session.userCase.caseTypeOfApplication) {
-          req.session.userCase?.respondents?.forEach((respondent: Respondent) => {
-            if (req.url.includes('support-you-need-during-case')) {
-              respondent.value.response['supportYouNeed'] = prepareRequest(req.session.userCase);
-            }
-            //}
-          });
-        } else {
-          if (req.url.includes('support-you-need-during-case')) {
-            req.session.userCase.respondentsFL401!.response['supportYouNeed'] = prepareRequest(req.session.userCase);
-          }
+      const { user, userCase } = req.session;
+      const partyType = getCasePartyType(userCase, user.id);
+      const partyDetails = getPartyDetails(userCase, user.id);
+      const client = new CosApiClient(user.accessToken, 'https://return-url');
+      if (partyDetails) {
+        Object.assign(partyDetails.response, { supportYouNeed: prepareRequest(userCase) });
+        req.session.userCase = await client.updateCaseData(
+          user,
+          userCase.id,
+          partyDetails,
+          partyType,
+          userCase.caseTypeOfApplication as CaseType,
+          CaseEvent.CITIZEN_CASE_UPDATE
+        );
+        mapDataInSession(req.session.userCase, user.id);
+        let return_url = RESPONDENT_TASK_LIST_URL;
+        if (partyType === PartyType.APPLICANT) {
+          return_url = APPLICANT_TASK_LIST_URL;
+        } else if (partyType === PartyType.RESPONDENT && req.session.userCase.caseTypeOfApplication === 'C100') {
+          return_url = RESPOND_TO_APPLICATION;
         }
-      } else if (req.url.includes('applicant')) {
-        if ('C100' === req.session.userCase.caseTypeOfApplication) {
-          req.session.userCase?.applicants?.forEach((applicant: Applicant) => {
-            if (applicant?.value?.user?.idamId === req.session?.user.id) {
-              if (req.url.includes('support-you-need-during-case')) {
-                applicant.value.response['supportYouNeed'] = prepareRequest(req.session.userCase);
-              }
-            }
-          });
+
+        if (partyType === PartyType.APPLICANT) {
```

Comment:
As discussed I told you to use from appSettings to redirect back.

Reply:
Yes Vivek, Done this.